linuxkit/linuxkit
# kubernetes project updates

This PR could be considered a WIP, in that there is plenty more to do, but I think it would be better to merge several tranches of changes rather than wait for the final result.

**- What I did**

Several things:
* Updated the `projects/kubernetes` build to use the more standard `pkg/package.mk` and `Dockerfile` based build scheme, dropping the use of the box tool. Cleaned up some of the previously semiauto generated scripts to become first class source (still more to do there). Image is in the `linuxkitprojects` org.
* Used image labels to simplify the yaml side of things.
* Updated the `boot.sh` to expose ssh port on the host (handy on Linux, although I'm not using the `ssh_into_kubelet.sh` script myself so I haven't touched it)
* Updated to the standard `pkg/mounts` image
* Disabled content trust for `gcr.io`, it has never supported it so the `image-cache` target can never have worked. It does now.

**- How I did it**

Emacs mostly.

**- How to verify it**

I've been using `boot.sh` on Linux, which suffices to check something boot but due to there being no network connectivity between individual QEMU VMs in the default setup I've also tested with two `virt-manager` vms.

**- Description for the changelog**
Updates and improvements to the Kubernetes project.

**- A picture of a cute animal (not mandatory but encouraged)**
[![](https://upload.wikimedia.org/wikipedia/en/7/79/Fallen_Angel_of_Doom.jpg "Blasphemy, Fallen Angel of Doom")](https://en.wikipedia.org/wiki/Fallen_Angel_of_Doom)
